File: 백준 Python/Step7 문자열 (단계번호 바뀜 후)/pro7_5 NO.1157.py
#아스키 코드를 사용하면 더 깔끔할까?? 알파벳 26개
s = input()
alphabet=["a","A","b","B","c","C","d","D","e","E","f","F","g","G","h","H","i","I","j","J","k","K","l","L","m","M","n","N","o","O","p","P","q","Q","r","R","s","S","t","T","u","U","v","V","w","W","x","X","y","Y","z","Z"]
count = [0]*26
maxi = -1
for i in range(len(s)) :
    for j in range(26) : #알파벳 26개와 비교
        if s[i] == alphabet[2*j] or s[i] == alphabet[2*j+1] :
            count[j] +=1
            #print(s[i],'의 인덱스는' ,j)
#print(["a","b","c","d","e","f","g","h","i","j","k","l","m","n","o","p","q","r","s","t","u","v","w","x","y","z"])
#print(count)
#print(max(count))
#print(count.index(max(count)))
maxi = count.index(max(count))
only = True
for i in range(26) :
    if max(count) == count[i] :
        if i != maxi :
            only = False
if only == False :
    print('?')
elif only == True :
    print(alphabet[2*maxi+1])
    

        

File: 백준 Java/Step7_new 문자열/src/Pro7_5_no1157.java
import java.io.*;
public class Pro7_5_no1157 {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		String s = br.readLine();
		String alphabet[] = {"a","A","b","B","c","C","d","D","e","E","f","F","g","G","h","H","i","I","j","J","k","K","l","L","m","M","n","N","o","O","p","P","q","Q","r","R","s","S","t","T","u","U","v","V","w","W","x","X","y","Y","z","Z"};
		int count[] = {0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0};
		for(int i=0; i<s.length(); i++) {
			for(int j=0; j<26; j++) {
				if(s.charAt(i)==alphabet[2*j].charAt(0) | s.charAt(i)==alphabet[2*j+1].charAt(0)) {
					count[j] += 1;
				}
			}
		}
		//count최댓값 찾기
		int maxi = 0;
		int max = count[0];
		for(int i=0; i<26; i++) {
			if (count[i]>max) {
				max = count[i];
				maxi = i;
			}
		}
		//최댓값 중복 검사
		boolean only = true;
		for(int i=0; i<26; i++) {
			if (max == count[i] && maxi != i) {
				only = false;
			}
		}
		//결과 출력
		if(only) {
			System.out.println(alphabet[2*maxi+1]);
		}
		else if(!only) {
			System.out.println("?");
			
		}
		
	}

}
